src/benchmarks/gc/GC.Infrastructure/GC.Analysis.API/DynamicEvents/DynamicEvents.cs

```diff
@@ -43,69 +43,85 @@ public sealed class DynamicEventSchema
 
         public int MaxOccurrence { get; init; } = 1;
 
-        public static void Set(List<DynamicEventSchema> dynamicEventSchemas, bool allowPartialSchema = true)
+        static DynamicEventSchema()
         {
-            DynamicEventSchemas.Clear();
-            DynamicEventSchema.allowPartialSchema = allowPartialSchema;
-            foreach (DynamicEventSchema dynamicEventSchema in dynamicEventSchemas)
+            // One time add.
+            if (DynamicEventSchemas.Count == 0)
+            {
+                // Add the adaptation schemas.
+                Add(GCDynamicEvents.SizeAdaptationSampleSchema);
+                Add(GCDynamicEvents.SizeAdaptationTuningSchema);
+            }
+        }
+
+        public static void Add(DynamicEventSchema dynamicEventSchema, bool allowPartialSchema = true)
```

Comment:
[existing] The use of a static method/field for this means that the GC infra dll can only be used for one dynamic event schema at a time.

Reply:
Yes - this is the intended behavior. We currently only want one set of dynamic event schemas. In what circumstance should we want multiple dynamic event schemas?

Reply:
Admittedly this was initially just "statics are bad" commentary.

Initial thing that I can think of is iterating on a dynamic event in a local build and wanting to keep a notebook running with an old run and a new one at the same time?

Or maybe if the infra library ended up embedded in some tool (or server for a tool) like the insights report where it might have multiple traces open concurrently.  Perhaps such a scenario wouldn't overlap with a local build changing the dynamic events.  Though what is a versioning story/rules for dynamic events?  Part of the point is that they can be changed, but these datas ones are getting special treatment - what if .net 9 and .net 10 have different forms for the datas events?

Reply:
I think a design like this might be appropriate.

- From the `GC.Analysis.API.dll`, we can expose a few public property that describe the dynamic event schema for each version, then
- When we open the trace, we can pick one from the list and associate it with the trace. That object from the public API is obviously just a template, we can modify it when we see fit (e.g. hacks on existing runtime with extra instrumentation)
- When `DynamicEvent()` is called, we know which trace it is, so we can figure out which schema to use for the decoding.

Is that something that will make us happy?